src/client/visualizations/line-chart/base-chart/background.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { scale } from "d3";
+import * as React from "react";
+import { Stage } from "../../../../common/models/stage/stage";
+import { Unary } from "../../../../common/utils/functional/functional";
+import { GridLines } from "../../../components/grid-lines/grid-lines";
+import { VerticalAxis } from "../../../components/vertical-axis/vertical-axis";
+import { ContinuousTicks } from "../utils/pick-x-axis-ticks";
+import { ContinuousScale } from "../utils/scale";
+import "./background.scss";
+
+interface BackgroundProps {
+  gridStage: Stage;
+  axisStage: Stage;
+  xScale: ContinuousScale;
+  xTicks: ContinuousTicks;
+  yScale: Linear;
+  formatter: Unary<number, string>;
+}
+
+type Linear = scale.Linear<number, number>;
+
+function calculateTicks(scale: Linear) {
+  return scale.ticks(5).filter((n: number) => n !== 0);
+}
+
+const TICK_WIDTH = 5;
```

Comment:
any reference to VerticalAxisProps.tickSize?

Reply:
Yes. But is used in two different places, so extracted here.